SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp

```diff
@@ -29,6 +34,45 @@ void UGlobalStateManager::ApplyData(const Worker_ComponentData& Data)
 	SingletonNameToEntityId = GetStringToEntityMapFromSchema(ComponentObject, 1);
 }
 
+void UGlobalStateManager::ApplyMapData(const Worker_ComponentData& Data)
+{
+	Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data.schema_type);
+	
+	// Set the Map URL.
+	if(Schema_GetObjectCount(ComponentObject, 1) == 1)
```

Comment:
If these schema queries are wrong, all bets are off right? Should they just be checks?

Reply:
Nah there are actually cases where it's possible to receive empty component data for an entity, for example if a component was never added to said entity in the snapshot. I've also had empty data come through when authority was changing over an entity so I believe it's best to just make the changes respective to whats in the data.